packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1164,22 +1164,26 @@ abstract class FlutterCommand extends Command<void> {
     );
   }
 
+  /// Returns a [FlutterProject] view of the current directory or a ToolExit error,
+  /// if `pubspec.yaml` or `example/pubspec.yaml` is invalid.
+  FlutterProject get project => FlutterProject.current();
+
   /// Compute the [BuildInfo] for the current flutter command.
+  ///
   /// Commands that build multiple build modes can pass in a [forcedBuildMode]
   /// to be used instead of parsing flags.
   ///
   /// Throws a [ToolExit] if the current set of options is not compatible with
   /// each other.
-  Future<BuildInfo> getBuildInfo({ BuildMode? forcedBuildMode, File? forcedTargetFile }) async {
+  Future<BuildInfo> getBuildInfo({BuildMode? forcedBuildMode, File? forcedTargetFile}) async {
     final bool trackWidgetCreation = argParser.options.containsKey('track-widget-creation') &&
       boolArg('track-widget-creation');
 
     final String? buildNumber = argParser.options.containsKey('build-number')
       ? stringArg('build-number')
       : null;
 
-    final File packagesFile = globals.fs.file(
-      packagesPath ?? globals.fs.path.absolute('.dart_tool', 'package_config.json'));
+    final File packagesFile = globals.fs.file(packagesPath ?? project.packageConfigFile);
```

Comment:
Consider renaming `packagesFile` to `packageConfigFile`.

Reply:
Done